docs/event-reducer.md

```diff
@@ -0,0 +1,58 @@
+# Using an Event Reducer
+
+By default, Dashbling keeps track of the last event data per event type. For advanced use cases you can customize this behaviour by providing an `EventReducer`.
+
+An `EventReducer` looks like this:
+
+```typescript
+interface Reducer {
+  (eventId: string, previousState: any | null, eventData: any): any;
+}
+```
+
+Event data that flows into Dashbling, either from jobs or from data pushed in via the REST API, is passed to the reducer allow it to modify the data before it gets stored in the history and pushed to clients.
+
+An `EventReducer` is a function that takes an `eventId`, the previous state that was stored for this event (or `null` if it's the first time an event is triggered) and the `eventData` that's coming in. The default implementation simply returns the new data like this:
+
+```javascript
+const defaultReducer = (_eventId, _previousState, eventData) => {
+  return eventData;
+};
+```
+
+You can create a custom reducer in case you want to override this behaviour.
+
+For example say you have events with the following data:
+
+```javascript
+const event = {
+  name: "Alice"  
+};
+```
+
+And you want to aggregate all events, you might define a reducer like this:
+
+```javascript
+eventReducer: (id, eventState, event) => {
```

Comment:
No, these are the function params: eventId, previous event state and the current event data.

Reply:
I think this is part of the dashbling config object. Perhaps it would be more clear if the function was extracted and referenced in the config object?

Reply:
I see, so does this mean that there is 1 global event reducer? Would be cool if we could split it up based on event id too! (Technically that is now already possible if we implement the switch statement ourselves I think..)

Reply:
Yes, indeed. We could even write our own `combineReducers`, like Redux.

Reply:
Yes it's part of the `dashbling.config.js`. I'll update the docs to clarify that.

And indeed it's inspired by Redux. You could indeed write something like Redux `combineReducer` or simply something like this:

```javascript
const reducers = {
  "eventA": require("./reducers/eventA"),
  "eventB": require("./reducers/eventB")
};

// dashbling.config.js
eventReducer: (id, eventState, event) => {
  reducers[id](eventState, event);
}

Reply:
`return` is missing, but it's a sane approach indeed 👍